### ISSUE TITLE
Karma - Chrome Headless - test FAILED - Uncaught null thrown

### ISSUE THREAD

**Issue by KissBalazs** [karma-runner/karma#3766]
Our unit tests fails in an undeterministic way, with an uninformative error log.

```
Chrome 89.0.4389.114 (Linux x86_64): Executed 1225 of 1453 SUCCESS (0 secs / 3 mins 29.829 secs)
Chrome 89.0.4389.114 (Linux x86_64): Executed 1226 of 1453 SUCCESS (0 secs / 3 mins 30.893 secs)
Chrome 89.0.4389.114 (Linux x86_64): Executed 1227 of 1453 SUCCESS (0 secs / 3 mins 31.923 secs)
Chrome 89.0.4389.114 (Linux x86_64) ChooseOfWorkplaceComponent should show institutie warning info FAILED
        Uncaught null thrown
Chrome 89.0.4389.114 (Linux x86_64): Executed 1228 of 1453 (1 FAILED) (0 secs / 3 mins 32.237 secs)
    ✗ should show institutie warning info
        Uncaught null thrown
Chrome 89.0.4389.114 (Linux x86_64): Executed 1229 of 1453 (1 FAILED) (0 secs / 3 mins 32.713 secs)
Chrome 89.0.4389.114 (Linux x86_64): Executed 1230 of 1453 (1 FAILED) (0 secs / 3 mins 33.432 secs)
Chrome 89.0.4389.114 (Linux x86_64): Executed 1231 of 1453 (1 FAILED) (0 secs / 3 mins 33.461 secs)
```

The example above is just one of the many different logs we get. We run the tests with the same order, but nevertheless the error happens on different test(s) all the time: `Uncaught null thrown`.

It happens with Angular 13, and only in one of many different microfrontend applications. We have over 1600 unit tests and it's impossible to narrow down to check which one causes it, as the error presents itself only sometimes... 

You can read a detailed description, jasmine config, different appriaches we tried in [this StackOverflow issue](https://stackoverflow.com/questions/71266939/karma-chrome-headless-test-failed-uncaught-null-thrown).  Thank you for your time for reading this! :)

**Comment by devoto13** [comment]
@KissBalazs I'm not an expert in this area, but it may actually work. I'm not sure if you really need `fakeAsync` (it seems to be intended for pretty special cases), but maybe if you wrap your tests into [`waitForAsync`](https://angular.io/api/core/testing/waitForAsync) it will help you localize the race as tests will wait until all async activities are done and therefore the error will be report in the same test which has innitiated the problematic async activity.

**Comment by KissBalazs** [comment]
To anyone who might stumble upon this: I was able to catch the error with the "stop on exception" feature of chrome. In it we saw that the error happens in a `zone.js` `task` object, which provided enough information about what causes the error.

It was us. Sorry for the trouble. We threw a `null` object in one of the tests and did not catch it. That's a quite bad practice. 

Sorry for taking up your time, and thank you again for the help.
